policies/step_annotations.rego

```diff
@@ -0,0 +1,25 @@
+package hacbs.contract.step_annotations
+
+import data.lib
+
+deny[{"msg": msg}] {
+	att := data.attestations[_]
+
+	some step_index
+
+	# null annotations show up when there are none defined
+	# can there be null annotations?
+	att.predicate.buildConfig.steps[step_index].annotations != null
```

Comment:
Not 100% sure it would work, but maybe you could instead put `null` in the list of allowed_annotations, to express more clearly the idea that the policy allows it. Then it's tidier and you don't need this code to handle it specially.

(IIUC this is different from the key being missing entirely. I think then the rule would quietly quit and we won't see a violation.)